se.cendio.tlclient.json

```diff
@@ -0,0 +1,58 @@
+{
+	"app-id": "se.cendio.tlclient",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "21.08",
+	"sdk": "org.freedesktop.Sdk",
+	"command": "tlclient-openconf",
+	"tags": [
+		"proprietary"
+	],
+	"finish-args": [
+		"--persist=.fltk",
+		"--persist=.thinlinc",
+		"--share=ipc",
+		"--share=network",
+		"--socket=x11",
+		"--socket=pulseaudio"
```

Comment:
Smart card support seems to need `--socket=pcsc`:

https://github.com/flatpak/flatpak/issues/4723

Reply:
there is no lpr. if you support cups properly it can be added.

Reply:
This issue has more smart card details, and unfortunately support is fragile:

https://github.com/flatpak/flatpak/issues/4066

Reply:
> there is no lpr. if you support cups properly it can be added.

Do you have some reference on what is provided? E.g. should cups libraries be included in the flatpak and there is some socket mapped?

Reply:
There's a `--socket=cups`, but I'm not sure if adding that is sufficient for it to work properly.

Reply:
I added the pcsc and cups sockets, but I am not sure how to test whether they work.

Reply:
I assume that cups and pcsc-lite needs to be built and put inside the flatpak. That seems to be what the linked issues do.

@Newbytee, you can test smart card support without an actual smart card. The client will say "No smart card available" if smart card authentication is enabled. But it will say "No smart card support" if PC/SC support is missing.

Likewise you should be able to test printing by setting up some dummy printer queue locally.

Reply:
> [@Newbytee](https://github.com/Newbytee), you can test smart card support without an actual smart card. The client will say "No smart card available" if smart card authentication is enabled. But it will say "No smart card support" if PC/SC support is missing.

Where would I see this?

![image](https://user-images.githubusercontent.com/4428753/159889374-adf648dc-1b2c-486a-b3b0-a1aeef08985a.png)


![image](https://user-images.githubusercontent.com/4428753/159889323-456e6264-5d5c-4d86-9893-c1b790176af7.png)

Reply:
If you close the options after switching to smart card authentication the password field will be replaced with a combo box with that message.

Reply:
Thanks. Unfortunately:

![image](https://user-images.githubusercontent.com/4428753/160083469-cbe6e517-b3db-4002-8bcd-7fa1f0efdab9.png)

Reply:
Yes, your local system needs to have some basic smart card support as well. pcsc-lite needs to be installed and pcscd needs to be running (or autostarted by systemd).

The ThinLinc client's log file might have some details as well.

Reply:
So we've tested here now, and it looks like the following need to be included in the app (or with some other runtime?):

 * `libpcsclite.so.1`
 * cups-client, and probably cups-libs

Reply:
Thanks, I'll look into that.

Reply:
As for lpr you may build just it instead of whole cups, see https://github.com/flathub/org.kde.okular/blob/404f59630e7d43bcddb97a76181e8f99f90422fd/org.kde.okular.json#L451-L470

Reply:
After adding  libpcsclite.so.1, PCSC seems to work on my system: :partying_face: 

![image](https://user-images.githubusercontent.com/4428753/161263250-20f2a301-3746-47ef-9ecb-58fdd9392f92.png)

I've also added lpr, but I'm not sure how to test whether it works.